src/transforms/rotation.jl

```diff
@@ -0,0 +1,70 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Rotation(rot)
+
+Rotate geometry or mesh with rotation `rot` from ReferenceFrameRotations.jl. 
+
+## Examples
+
+```julia
+Rotation(ReferenceFrameRotations.EulerAngleAxis(pi/4, [1, 0, 0]))
+```
+
+"""
+struct Rotation{R} <: GeometricTransform
+  rot::R
+end
+
+"""
+    Rotation(axis, α; degrees=true)
+
+Perform the rotation of angle `α` around the axis directed by the 
+vector `axis`. The angle must be given in degrees if `degrees=true`, 
+otherwise in radians.
+
+"""
+function Rotation(axis::Vector{Real}, α::Real; degrees::Bool = true)
+  axisNorm = norm2(axis)
+  if axisNorm == 0
+    error("The `axis` vector is null.")
+  end
+  v = axis / axisNorm
+  if degrees
+    α = α * pi / 180
+  end
+  Rotation(EulerAngleAxis(α, v))
+end
+
+isrevertible(::Type{<:Rotation}) = true
+
+function preprocess(transform::Rotation, object)
+  rot = transform.rot
+  local M, M⁻¹
```

Comment:
No need to use the keyword `local`, it is rarely needed in Julia code.

Reply:
Even if `M` is introduced in a `if..else`? Strange, I was sure this was needed.

Reply:
A more idiomatic Julia code would define variables like:

```julia
var = if condition
  val1
else
  val2
end
```

Reply:
or even:

```julia
var = condition ? val1 : val2
```